File: code/api/src/Models/Database/Project/Project.cs
namespace IOL.GreatOffice.Api.Data.Database;

public class Project : BaseWithOwner
{
    public Project() { }

    public Project(LoggedInUserModel loggedInUserModel) : base(loggedInUserModel) { }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime? Start { get; set; }
    public DateTime? Stop { get; set; }
    public ICollection<Customer> Customers { get; set; }
    public ICollection<ProjectMember> Members { get; set; }
    public ICollection<ProjectLabel> Labels { get; set; }
}

File: code/frontend/src/models/projects/Project.ts
import type { Temporal } from "temporal-polyfill"
import type { ProjectMember } from "./ProjectMember"
import type { ProjectStatus } from "./ProjectStatus"

export type Project = {
    id: string,
    name: string,
    description?: string,
    start: Temporal.PlainDate,
    stop?: Temporal.PlainDate,
    members: Array<ProjectMember>,
    status: ProjectStatus
}